Adds glossary page for clickjacking

KanishkTagade opened the pull request:
This PR adds a page for the Clickjacking entry in Glossary. 

As requested by the moderator, made all changes within one branch, mentioned here: https://github.com/mdn/content/pull/8577

KanishkTagade committed:
Creating index.html within Clickjacking folder

Adding Clickjacking to Glossary

KanishkTagade committed:
Added MDN page link for Clickjacking

Replaced existing links with MDN Glossary Link for Clickjacking

KanishkTagade committed:
Changing file type index.html to index.md

estelle committed:
space and case

estelle commented on files/en-us/learn/server-side/apache_configuration_htaccess/index.md:
```diff
@@ -473,7 +473,7 @@ Be aware that Strict Transport Security is not revokable and you must ensure bei
 4.  Prevents all websites (including your own) from embedding your webpages within e.g. the `<iframe>` or `<object>` element by setting: `frame-ancestors 'none'`.
 
-    - The `frame-ancestors`directive helps avoid "Clickjacking" attacks and is similar to the `X-Frame-Options` header
+    - The `frame-ancestors`directive helps avoid [Clickjacking](/en-us/Glossary/Clickjacking) attacks and is similar to the `X-Frame-Options` header
```
Please make clickjacking lowercase in all cases where it should be lowercase in the rest of this PR. Should it be quoted here?

estelle commented on files/en-us/learn/server-side/django/web_application_security/index.md:
```diff
@@ -130,7 +130,7 @@ Django also provides other forms of protection (most of which would be hard or n
   - : SQL injection vulnerabilities enable malicious users to execute arbitrary SQL code on a database, allowing data to be accessed, modified, or deleted irrespective of the user's permissions. In almost every case you'll be accessing the database using Django’s querysets/models, so the resulting SQL will be properly escaped by the underlying database driver. If you do need to write raw queries  ...
 - Clickjacking protection
-  - : In this attack a malicious user hijacks clicks meant for a visible top level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials in an invisible [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame Element (<iframe>) represents a nested browsing context, effectively e ...
+  - : In this attack a malicious user hijacks clicks meant for a visible top level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials in an invisible [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame Element (<iframe>) represents a nested browsing context, effectively e ...
```
the glossary doesn't really talk about clickjacking "protection". A few options: only link the clickjacking word. Link clickjacking to glossary and protection to the django article, or revert this change.

estelle commented on files/en-us/glossary/clickjacking/index.md:
Unwarily is used correctly, but I think "unwittingly" may be easier to understand ... or something more understandable even.

> In this attack, a malicious user hijacks clicks meant for a visible top-level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials into an invisible {{htmlelement("iframe")}} controlled by the attacker. Clickjacking could also be used to get the user to click a button on a visible site, but in doing so actually unwittingly click a completely different button. As a defense, your site can prevent itself from being embedded in an iframe in another site by setting the appropriate HTTP headers.

marcelozarate replied:
unintentionally maybe?

estelle commented on files/en-us/learn/server-side/first_steps/website_security/index.md:
```diff
@@ -121,7 +121,7 @@ Web frameworks often include such CSRF prevention mechanisms.
 Other common attacks/vulnerabilities include:
 
-- [Clickjacking](https://www.owasp.org/index.php/Clickjacking). In this attack, a malicious user hijacks clicks meant for a visible top-level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials into an invisible {{htmlelement("iframe")}} controlled by the attacker. Clickjacking could also be  ...
+- [Clickjacking](/en-us/Glossary/Clickjacking). In this attack, a malicious user hijacks clicks meant for a visible top-level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials into an invisible {{htmlelement("iframe")}} controlled by the attacker. [Clickjacking](/en-us/Glossary/Clickjacking ...
```
Nice definition! Uppercase ok here

estelle replied:
```suggestion
- [Clickjacking](/en-us/Glossary/Clickjacking). In this attack, a malicious user hijacks clicks meant for a visible top-level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials into an invisible {{htmlelement("iframe")}} controlled by the attacker. Clickjacking could also be used to get the user to click a button on a visible site, but in doing so actually unwittingly click a completely different button. As a defense, your site can prevent itself from being embedded in an iframe in another site by setting the appropriate HTTP headers.
```

estelle reviewed (changes requested):
Clickjacking should be capitalized only at start of sentences. Glossary needs to be in markdown rather than HTML. Nice addition to the glossary!

KanishkTagade committed:
Adds link to only "Clickjacking" word

Change suggested here: https://github.com/mdn/content/pull/10226/files#r741459564

KanishkTagade reviewed (commented):
Hello @estelle, Thank you very much for your detailed review and suggestions. I have checked and corrected the suggested changes and marked them as "Viewed" here. 

I am not sure if I can resolve the conversations here (I purposely left them for another reviewer). Thank you again.
 
I am excited about my first MDN glossary entry. 😃

KanishkTagade commented:
Hello @estelle and @marcelozarate . Apologies for following up on this after a long time. Would you please guide me on the next steps required for adding "Clickjacking" entry to the MDN Glossary? Thank you :)

estelle committed:
remove spam

estelle commented on files/en-us/learn/server-side/django/introduction/index.md:
```diff
@@ -61,7 +61,7 @@ Django helps you write software that is:
     _A password hash is a fixed-length value created by sending the password through a [cryptographic hash function](https://en.wikipedia.org/wiki/Cryptographic_hash_function). Django can check if an entered password is correct by running it through the hash function and comparing the output to the stored hash value. However due to the "one-way" nature of the function, even if a stored hash value ...
 
-    Django enables protection against many vulnerabilities by default, including SQL injection, cross-site scripting, cross-site request forgery and clickjacking (see [Website security](/en-US/docs/Learn/Server-side/First_steps/Website_security) for more details of such attacks).
+    Django enables protection against many vulnerabilities by default, including SQL injection, cross-site scripting, cross-site request forgery and [Clickjacking](/en-us/Glossary/Clickjacking) (see [Website security](/en-US/docs/Learn/Server-side/First_steps/Website_security) for more details of such attacks).
```
```suggestion
    Django enables protection against many vulnerabilities by default, including SQL injection, cross-site scripting, cross-site request forgery and [clickjacking](/en-us/Glossary/Clickjacking) (see [Website security](/en-US/docs/Learn/Server-side/First_steps/Website_security) for more details of such attacks).
```

estelle commented on files/en-us/learn/server-side/django/web_application_security/index.md:
```diff
@@ -130,7 +130,7 @@ Django also provides other forms of protection (most of which would be hard or n
   - : SQL injection vulnerabilities enable malicious users to execute arbitrary SQL code on a database, allowing data to be accessed, modified, or deleted irrespective of the user's permissions. In almost every case you'll be accessing the database using Django’s querysets/models, so the resulting SQL will be properly escaped by the underlying database driver. If you do need to write raw queries  ...
 - Clickjacking protection
-  - : In this attack a malicious user hijacks clicks meant for a visible top level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials in an invisible [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame Element (<iframe>) represents a nested browsing context, effectively e ...
+  - : In this attack a malicious user hijacks clicks meant for a visible top level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials in an invisible [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame Element (<iframe>) represents a nested browsing context, effectively e ...
```
```suggestion
  - : In this attack a malicious user hijacks clicks meant for a visible top level site and routes them to a hidden page beneath. This technique might be used, for example, to display a legitimate bank site but capture the login credentials in an invisible [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame Element (<iframe>) represents a nested browsing context, effectively embedding another HTML page into the current page. In HTML 4.01, a document may contain a head and a body or a head and a frameset, but not both a body and a frameset. However, an <iframe> can be used within a normal document body. Each browsing context has its own session history and active document. The browsing context that contains the embedded content is called the parent browsing context. The top-level browsing context (which has no parent) is typically the browser window.") controlled by the attacker. Django contains [clickjacking](/en-us/Glossary/Clickjacking) protection in the form of the [`X-Frame-Options middleware`](https://docs.djangoproject.com/en/3.1/ref/middleware/#django.middleware.clickjacking.XFrameOptionsMiddleware "django.middleware.clickjacking.XFrameOptionsMiddleware") which, in a supporting browser, can prevent a site from being rendered inside a frame.
```

estelle commented on files/en-us/web/security/index.md:
```diff
@@ -65,10 +65,10 @@ Ensuring that your website or open web application is secure is critical. Even s
 ## Clickjacking protection
 
-In clickjacking, a user is fooled into clicking on a UI element that performs some action other than what the user expects.
+In [Clickjacking](/en-us/Glossary/Clickjacking), a user is fooled into clicking on a UI element that performs some action other than what the user expects.
```
```suggestion
In [clickjacking](/en-us/Glossary/Clickjacking), a user is fooled into clicking on a UI element that performs some action other than what the user expects.
```

estelle commented on files/en-us/web/security/index.md:
```diff
@@ -65,10 +65,10 @@ Ensuring that your website or open web application is secure is critical. Even s
 
 - [HTTP X-Frame-Options](/en-US/docs/Web/HTTP/Headers/X-Frame-Options)
-  - : The **`X-Frame-Options`** [HTTP](/en-US/docs/Web/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a [`<frame>`](/en-US/docs/Web/HTML/Element/frame "<frame> is an HTML element which defines a particular area in which another HTML document can be displayed. A frame should be used within a <frameset>."), [`<iframe>`](/en-US/docs/Web/H ...
+  - : The **`X-Frame-Options`** [HTTP](/en-US/docs/Web/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a [`<frame>`](/en-US/docs/Web/HTML/Element/frame "<frame> is an HTML element which defines a particular area in which another HTML document can be displayed. A frame should be used within a <frameset>."), [`<iframe>`](/en-US/docs/Web/H ...
```
```suggestion
  - : The **`X-Frame-Options`** [HTTP](/en-US/docs/Web/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a [`<frame>`](/en-US/docs/Web/HTML/Element/frame "<frame> is an HTML element which defines a particular area in which another HTML document can be displayed. A frame should be used within a <frameset>."), [`<iframe>`](/en-US/docs/Web/HTML/Element/iframe "The HTML Inline Frame element (<iframe>) represents a nested browsing context, embedding another HTML page into the current one."), [`<embed>`](/en-US/docs/Web/HTML/Element/embed "The HTML <embed> element embeds external content at the specified point in the document. This content is provided by an external application or other source of interactive content such as a browser plug-in.") or [`<object>`](/en-US/docs/Web/HTML/Element/object "The HTML <object> element represents an external resource, which can be treated as an image, a nested browsing context, or a resource to be handled by a plugin."). Sites can use this to avoid [clickjacking](/en-us/Glossary/Clickjacking) attacks, by ensuring that their content is not embedded into other sites.
```

estelle commented on files/en-us/web/security/securing_your_site/index.md:
```diff
@@ -33,7 +33,7 @@ There are a number of things you can do to help secure your site. This article o
   - : An added layer of security that helps to detect and mitigate certain types of attacks, including {{Glossary("Cross-site_scripting", "Cross Site Scripting (XSS)")}} and data injection attacks. These attacks are used for everything from data theft to site defacement or distribution of malware. Code is executed by the victims and lets the attackers bypass access controls and impersonate users. ...
 - [The X-Frame-Options response header](/en-US/docs/Web/HTTP/X-Frame-Options)
-  - : The `X-Frame-Options:` [HTTP](/en/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a {{ HTMLElement("frame") }}. Sites can use this to avoid clickjacking attacks, by ensuring that their content is not embedded into other sites.
+  - : The `X-Frame-Options:` [HTTP](/en/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a {{ HTMLElement("frame") }}. Sites can use this to avoid [Clickjacking](/en-us/Glossary/Clickjacking) attacks, by ensuring that their content is not embedded into other sites.
```
```suggestion
  - : The `X-Frame-Options:` [HTTP](/en/HTTP) response header can be used to indicate whether or not a browser should be allowed to render a page in a {{ HTMLElement("frame") }}. Sites can use this to avoid [clickjacking](/en-us/Glossary/Clickjacking) attacks, by ensuring that their content is not embedded into other sites.
```

estelle committed:
Apply suggestions from code review

lowercase clickjacking where appropriate

estelle committed:
converted from html to md

estelle committed:
word

estelle committed:
link
